### ISSUE TITLE
Resolve SHMEMX usage in CXX generic tests

### ISSUE THREAD

**Issue by jdinan** [Sandia-OpenSHMEM/SOS#800]
## Issue

C++ generic tests were classified as shmemx tests in #797.  However, these tests also test the nonblocking fetching atomics, which are extensions to the extensions.  Presently, both the C++ generic tests and the NBI testing within these tests are gated by the same automake flag `SHMEMX_TESTS`, so there is presently no way to enable the C++ generic tests without also enabling the NBI atomics cases in these tests.

## Proposed Fix

The NBI atomics should be standardized in OpenSHMEM 1.5, at which point we can delete all of the code in the tests that enables/disables these NBI atomics cases based on the value of `SHMEMX_TESTS`.  In this scenario, this issue can be closed once the relevant code is deleted from the C++ generic tests.  If NBI atomics don't go into OpenSHMEM 1.5, then we will need a more creative fix.

**Comment by jdinan** [comment]
Closed by #809

**Comment by davidozog** [comment]
@jdinan - we might need a little help understanding this issue.  It seems like we still may need to remove the `ENABLE_SHMEMX_TEST` around the C++ NBI AMOs.  Is that correct?

 These AMO (blocking and non-blocking) tests with the `.cpp` extension in `test/shmemx` should remain there, correct?

**Comment by jdinan** [comment]
I think the only thing to do here for the 1.5 release is to rename all of the bindings from shmemx -> shmem and update the tests to use the new functions. We made the choice to file all of the C++ tests under the shmemx directory (even though they use the shmem function prefix) since the C++ polymorphic bindings are not part of the OpenSHMEM specification.

**Comment by davidozog** [comment]
@jdinan @wrrobin Yeah, sounds good.

The thing that still confuses me slightly is that `ENABLE_SHMEMX_TESTS` wraps only the NBI atomics in the CXX "shmemx" tests, like [here](https://github.com/Sandia-OpenSHMEM/SOS/pull/955/files#diff-48864a7a268ddb49a5072d6f72b0f93bR54) and [here](https://github.com/Sandia-OpenSHMEM/SOS/pull/955/files#diff-48864a7a268ddb49a5072d6f72b0f93bR221), and that's true for all the NBI AMOs:
```
cxx_test_shmem_atomic_add.cpp 
cxx_test_shmem_atomic_and.cpp 
cxx_test_shmem_atomic_cswap.cpp 
cxx_test_shmem_atomic_fetch.cpp 
cxx_test_shmem_atomic_inc.cpp 
cxx_test_shmem_atomic_or.cpp 
cxx_test_shmem_atomic_swap.cpp 
cxx_test_shmem_atomic_xor.cpp
```

I think those should be removed now that the NBI AMOs are standardized, right?  It's just a little strange because I think removing these won't actually build anything differently (based on my understanding of Makefile.am).  But there's no need for the check anymore, since the NBI AMOs are of the same class as the blocking AMOs in v1.5.

Assuming ya'll agree - @rdesai16, do you want to remove those `ENABLE_SHMEMX_TESTS` `ifdef`'s in those files and push to PR #955?

**Comment by rdesai16** [comment]
@davidozog Yeah sure. I'll do that.

**Comment by davidozog** [comment]
Closed by #1030
